.github/pull_request_template.md

```diff
@@ -17,6 +17,8 @@ Resolves #[Issue number to be closed when this PR is merged]
   - [ ] All tests pass
 - [ ] New functionality has been documented.
   - [ ] New functionality has javadoc added
+- [ ] Specifications for new APIs to support generated clients have been added (See: [Guide](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md))
```

Comment:
Lets we make this a single checkbox item, similar to the docs issue at the bottom of this checklist.  What do you think?

```suggestion
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
```

Reply:
I have no strong opinions on this. The only thing is that we need to make sure if there is a need to update the API spec, there should be a way for the `API Spec` maintainers to catching this, like the doc team does.